### ISSUE TITLE
pip install returns multiple "Cache entry deserialization failed, entry ignored"

### ISSUE THREAD

**Issue by Deena-B** [biocommons/hgvs#540]
When I install hgvs I get a lot of messages that say "Cache entry deserialization failed, entry ignored" and one message that says "OSError: [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/ipython_genutils' ". Please see output below. Do any of these matter for running hgvs?  

I'm installing into an otherwise empty conda environment. 
I am using anaconda (version 1.7.2) on a macOS High Sierra (version 10.13.6)

$ `pip install hgvs`

```
Collecting hgvs
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/e9/84/3fb6741da05bc6fb7d397d23c8ddf9e5e11f068fd6e2f4c0cf547d94f9e8/hgvs-1.2.4-py2.py3-none-any.whl (85kB)
    100% |████████████████████████████████| 92kB 1.7MB/s
Collecting biocommons.seqrepo (from hgvs)
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/59/75/4e0e906e9df32e45f50341602d5dec3497a13d136ff6876a8713fc613fb7/biocommons.seqrepo-0.4.4-py2.py3-none-any.whl
Collecting psycopg2-binary (from hgvs)
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/13/02/f0867c90062d75f304790ec0061be9139eb2375da62d93e57ad44640e590/psycopg2_binary-2.7.6.1-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (1.5MB)
    100% |████████████████████████████████| 1.5MB 736kB/s
Collecting ipython<6 (from hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/b0/88/d996ab8be22cea1eaa18baee3678a11265e18cf09974728d683c51102148/ipython-5.8.0-py2-none-any.whl (760kB)
    100% |████████████████████████████████| 768kB 1.3MB/s
Collecting parsley (from hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/2b/d6/4fed8d65e28a970e1c5cb33ce9c7e22e3de745e1b2ae37af051ef16aea3b/Parsley-1.3-py2.py3-none-any.whl (88kB)
    100% |████████████████████████████████| 92kB 5.6MB/s
Collecting attrs>=17.4.0 (from hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/3a/e1/5f9023cc983f1a628a8c2fd051ad19e76ff7b142a0faf329336f9a62a514/attrs-18.2.0-py2.py3-none-any.whl
Collecting enum34 (from hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting unicodecsv (from hgvs)
  Cache entry deserialization failed, entry ignored
Collecting bioutils>=0.2.2 (from hgvs)
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/6b/f0/7708222c477423463aa453a4c657d7668fb2a52dcf9655d757b66f2b6f6e/bioutils-0.4.0-py2.py3-none-any.whl (253kB)
    100% |████████████████████████████████| 256kB 1.9MB/s
Requirement already satisfied: six in /Library/Python/2.7/site-packages (from hgvs)
Collecting configparser>=3.3.0 (from hgvs)
  Cache entry deserialization failed, entry ignored
Collecting biopython==1.69 (from hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/34/ad/65ee53813837f04d7ea02f66b93d90a02e06a410feb10413f7dddece3dca/biopython-1.69.tar.gz (15.4MB)
    100% |████████████████████████████████| 15.4MB 82kB/s
Requirement already satisfied: tqdm in /Library/Python/2.7/site-packages (from biocommons.seqrepo->hgvs)
Collecting pysam (from biocommons.seqrepo->hgvs)
  Cache entry deserialization failed, entry ignored
Collecting yoyo-migrations<6,>5 (from biocommons.seqrepo->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/03/e0/9f68c08ef31350eaf39fefa04b8fa69867c3b09999d08999ff90f16fbded/yoyo_migrations-5.1.7-py2.py3-none-any.whl
Collecting traitlets>=4.2 (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/93/d6/abcb22de61d78e2fc3959c964628a5771e47e7cc60d53e9342e21ed6cc9a/traitlets-4.3.2-py2.py3-none-any.whl (74kB)
    100% |████████████████████████████████| 81kB 5.4MB/s
Requirement already satisfied: setuptools>=18.5 in /System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python (from ipython<6->hgvs)
Collecting simplegeneric>0.8 (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
Collecting backports.shutil-get-terminal-size; python_version == "2.7" (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/7d/cd/1750d6c35fe86d35f8562091737907f234b78fdffab42b29c72b1dd861f4/backports.shutil_get_terminal_size-1.0.0-py2.py3-none-any.whl
Collecting pygments (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/13/e5/6d710c9cf96c31ac82657bcfb441df328b22df8564d58d0c4cd62612674c/Pygments-2.3.1-py2.py3-none-any.whl (849kB)
    100% |████████████████████████████████| 849kB 947kB/s
Collecting appnope; sys_platform == "darwin" (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/87/a9/7985e6a53402f294c8f0e8eff3151a83f1fb901fa92909bb3ff29b4d22af/appnope-0.1.0-py2.py3-none-any.whl
Collecting pexpect; sys_platform != "win32" (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/89/e6/b5a1de8b0cc4e07ca1b305a4fcc3f9806025c1b651ea302646341222f88b/pexpect-4.6.0-py2.py3-none-any.whl (57kB)
    100% |████████████████████████████████| 61kB 4.4MB/s
Collecting pickleshare (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/9a/41/220f49aaea88bc6fa6cba8d05ecf24676326156c23b991e80b3f2fc24c77/pickleshare-0.7.5-py2.py3-none-any.whl
Collecting prompt-toolkit<2.0.0,>=1.0.4 (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/d1/b0/1a6c262da35c779dd79550137aa7c298a424987240a28792ec5ccf48f848/prompt_toolkit-1.0.15-py2-none-any.whl (247kB)
    100% |████████████████████████████████| 256kB 2.0MB/s
Collecting decorator (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/bc/bb/a24838832ba35baf52f32ab1a49b906b5f82fb7c76b2f6a7e35e140bac30/decorator-4.3.0-py2.py3-none-any.whl
Collecting pathlib2; python_version == "2.7" or python_version == "3.3" (from ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/2a/46/c696dcf1c7aad917b39b875acdc5451975e3a9b4890dca8329983201c97a/pathlib2-2.3.3-py2.py3-none-any.whl
Requirement already satisfied: requests in /Library/Python/2.7/site-packages (from bioutils>=0.2.2->hgvs)
Collecting text-unidecode (from yoyo-migrations<6,>5->biocommons.seqrepo->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/79/42/d717cc2b4520fb09e45b344b1b0b4e81aa672001dd128c180fabc655c341/text_unidecode-1.2-py2.py3-none-any.whl (77kB)
    100% |████████████████████████████████| 81kB 3.7MB/s
Collecting iniherit (from yoyo-migrations<6,>5->biocommons.seqrepo->hgvs)
  Cache entry deserialization failed, entry ignored
Collecting ipython-genutils (from traitlets>=4.2->ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/fa/bc/9bd3b5c2b4774d5f33b2d544f1460be9df7df2fe42f352135381c347c69a/ipython_genutils-0.2.0-py2.py3-none-any.whl
Collecting ptyprocess>=0.5 (from pexpect; sys_platform != "win32"->ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/d1/29/605c2cc68a9992d18dada28206eeada56ea4bd07a239669da41674648b6f/ptyprocess-0.6.0-py2.py3-none-any.whl
Collecting wcwidth (from prompt-toolkit<2.0.0,>=1.0.4->ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/7e/9f/526a6947247599b084ee5232e4f9190a38f398d7300d866af3ab571a5bfe/wcwidth-0.1.7-py2.py3-none-any.whl
Collecting scandir; python_version < "3.5" (from pathlib2; python_version == "2.7" or python_version == "3.3"->ipython<6->hgvs)
  Cache entry deserialization failed, entry ignored
Installing collected packages: ipython-genutils, decorator, enum34, traitlets, simplegeneric, backports.shutil-get-terminal-size, pygments, appnope, ptyprocess, pexpect, scandir, pathlib2, pickleshare, wcwidth, prompt-toolkit, ipython, bioutils, pysam, text-unidecode, iniherit, yoyo-migrations, biocommons.seqrepo, psycopg2-binary, parsley, attrs, unicodecsv, configparser, biopython, hgvs
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/wheel.py", line 316, in clobber
    ensure_dir(destdir)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/utils/__init__.py", line 83, in ensure_dir
    os.makedirs(path)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/Library/Python/2.7/site-packages/ipython_genutils'
You are using pip version 9.0.1, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

**Comment by reece** [comment]
No. This is an innocuous error in pip that is annoying many people in many contexts.

The first Google hit when [searching for "Cache entry deserialization failed, entry ignored"](https://www.google.com/search?q=Cache+entry+deserialization+failed%2C+entry+ignored) leads to [this comment in pip](https://github.com/pypa/pip/issues/5250#issuecomment-394706450):

> I found that on Python 2, pip is able to use a cache populated by a Python 3 run but not vice versa.
> 
> A response cached on Py2 results in the cache storing a header that loads on Py3 as bytes instead of str. This causes the deserialization to fail. Working on a PR to fix it upstream.
> 
>

**Comment by Deena-B** [comment]
@reece Thanks for calling me on this. I restored the original subject and text and posted my new issue, which resolves this issue here: https://github.com/biocommons/hgvs/issues/543

(Sorry, I didn't see that you had already replied to this issue)

The fix for this issue (#540) was to install pip inside my virtual env with 
`conda install pip`

I also installed postgresql with
`conda install -c anaconda postgresql`

Then `pip install hgvs` worked
